doc/tapscript_opcodes.md

```diff
@@ -12,82 +12,72 @@ Taproot already increases a lot of resource limitations from segwitv0, so there
-      10. Otherwise treat as `OP_SUCCESS` and return true (without executing rest of script).
-   - Define `OP_SUCCESS200` as `OP_INSPECTCURRENTINPUT` that pushes the current input index(4) as little-endian onto the stack
-   - Define `OP_SUCCESS201` as `OP_INSPECTOUTPUT` with the following semantics:
-      - Pop the stack pop as minimal `CScriptNum` as `n`. Next, pop another element as minimal `CScriptNum` input index `idx`.
-      1. If `n=0`, Push the `nAsset` as a tuple, first push the assetID(32), followed by the prefix(1)
-      2. If `n=1`, Push the `nValue` as a tuple, value(8, 32) followed by prefix
-      3. If `n=2`, Push the `nNonce` as a tuple, nonce(32, 0) followed by prefix. Push empty vector for `None` nonce
-      4. If `n=3`, Push the scriptPubkey(35).
-      5. Otherwise treat as `OP_SUCCESS` and return true (without executing rest of script).
-   - Define `OP_SUCCESS202` as `OP_INSPECTTX` with the following semantics:
-      - Pop the stack pop as minimal `CScriptNum` as `n`.
-      1. If `n=0`, Push the nVersion(4) as little-endian.
-      2. If `n=1`, Push the nLockTime(4) as little-endian.
-      3. If `n=2`, Push the number of inputs(4) as little-endian
-      4. If `n=3`, Push the number of outputs(4) as little-endian
-      5. If `n=4`, Push the transaction size in vbytes (4) as little-endian
-      6. Otherwise treat as `OP_SUCCESS` and return true (without executing rest of script).
-
-5. **Crypto**: In order to allow more complex operations on elements, we introduce the following new crypto-operators.
-   - Define `OP_SUCCESS203` as `OP_ECMULSCALAREXPVERIFY`, pop the top element as `k`. Then pop next elements as points `G1`(first), `G2`(second)
-      ..`Gk`. Next pop `k` scalars `x1`, `x2`... `xk`. Finally pop result element as as point `Q`. Assert `x1G1+x2G2+x3G3.. xkGk == Q` This counts as `k*50` towards budget sigops. If any of `G_i`,`Q` is point at infinity fail.
-   - Define `OP_SUCCESS204` as `OP_TAPTWEAK` with the following semantics. Pop the first element as point `P`, second element as script blob `S`. Push the Taptweak on the top of stack `Q = P + H(P||S)*G`. If `|S| > MAX_ELEMENT_SIZE`, the user should use the streaming opcodes to compute the Hash function.
+For opcodes that inspect data that is not committed in sighash, introspection is safe because any changes to the witness data would cause wtxid to change and it would revalidate the tx again. For pegin inputs, the asset/value/script information will be one from the parent chain.
+   - Transaction input introspection opcodes:
+      1. Define `OP_SUCCESS199` as `OP_INSPECTINPUTOUTPOINT`: Pop a `CScriptNum` input index `idx` and push the outpoint as a tuple. First push the `txid`(32) of the `prev_out`, followed by a 4 byte push of `vout` followed by a pushed for push the outpoint_flag(1) as defined in [Modified BIP-341 SigMsg for Elements](https://gist.github.com/roconnor-blockstream/9f0753711153de2e254f7e54314f7169).
```

Comment:
Same here: the 4 byte push of the vout is little endian.

Typo: `followed by a pushed for push`